youtube.go

```diff
@@ -109,23 +110,56 @@ func (y *Youtube) StartDownloadWithQuality(destFile string, quality string) erro
 func (y *Youtube) StartDownloadFile() error {
 	//download highest resolution on [0]
 	err := errors.New("Empty stream list")
-	for _, v := range y.StreamList {
-		url := v["url"]
-		y.log(fmt.Sprintln("Download url=", url))
+	for _, stream := range y.StreamList {
+		streamUrl := stream["url"]
+		streamType := stream["type"]
+		y.log(fmt.Sprintln("Download url=", streamUrl))
+
+		// Find out what the file name should be.
+		fileName := sanitizeFilename(stream["title"])
+
+		// Find out what the file extension should be.
+		fileExtensions, err := mime.ExtensionsByType(streamType)
+		if err != nil {
+			fileName += ".mov"
+		} else {
+			fileName += fileExtensions[0]
+		}
```

Comment:
This is a new feature.

Reply:
It's a very good feature.
 just have small question about it, why do you choose m4v,  as shown as in the picture below? because I think mp4 is more common.

![image](https://user-images.githubusercontent.com/7224544/80030515-98db2800-84e8-11ea-9136-3eb5aff47db7.png)

Reply:
A bad assumption. Often the mime list has only one known file extension.

You're right that for the mp4 format, the .mp4 extension is way more common than the other ones... not sure what Go is up to there.

I'll improve the detection by prioritizing the names with how close to the extension is to the mime type text.

Reply:
There, I changed the way the extension is determined. The ones from Go are too minimal and the ones from the operating system are too vague. 

I hope this is a decent compromise.